interface/src/app/volcano-plot/volcano-plot.js

```diff
@@ -4,50 +4,134 @@
  * via plotData.
  */
 angular.module('adage.volcano-plot', [
+  'ngResource',
   'ngVega',
   'greenelab.stats',
+  'adage.utils',
   'adage.volcano-plot-vgspec'
 ])
 
-.controller('VolcanoPlotCtrl', ['$scope', 'VolcanoPlotSpec',
-  function VolcanoPlotCtrl($scope, VolcanoPlotSpec) {
-    // define controller instance variables that link with vega
-    this.spec = VolcanoPlotSpec;
-    this.data = $scope.plotData;
-    // FIXME: can't get selectedNodes out this way... need to use view api?
-    // $scope.plotData.selectedNodes: $scope.selectedNodes
+// TODO when refactoring for issue #76, this service belongs in the top-level
+//      src/app/ module called `node` in its `services.js` file
+.factory('NodeService', ['$resource', 'ApiBasePath',
+  function($resource, ApiBasePath) {
+    // private methods for NodeService
+    var Participation = $resource(ApiBasePath + 'participation/');
+
+    // public methods for NodeService
+    return {
+      getGenesForNodesPromise: function getGenesForNodesPromise(nodes) {
+        var promise = Participation.get(
+          {'node__in': nodes.join(), 'limit': 0}
+        ).$promise.then(
+          function success(value) {
+            // when the data come back, condense the reponse into a flat list
+            // of gene IDs with no repeated entries
+            var geneList = value.objects.reduce(function reduce(acc, val) {
+              if (acc.indexOf(val.gene.pk) === -1) {
+                acc.push(val.gene.pk);
+              }
+              return acc;
+            }, []);
+            return geneList;
+          }
+        );
+        return promise;
+      }
+    };
   }
 ])
 
-.directive('volcanoPlot', function() {
-  return {
-    restrict: 'E',
-    scope: {
-      plotData: '=',
-        // plotData is a json list with the following format:
-        // [
-        //   {
-        //     'node': "<node-name-1>",
-        //     'diff': "<difference-in-mean-activity>",
-        //     'logsig': "<-log10(p-value-of-difference)>",
-        //   },
-        //   {...repeat for each node...}
-        // ]
-      selectedNodes: '='
-        // selectedNodes is a json list with the following format:
-        // [
-        //   {
-        //     'node': "<node-name-1>"
-        //   },
-        //   {
-        //     'node': "<node-name-2>"
-        //   },
-        //   {...repeat for each node...}
-        // ]
-    },
-    templateUrl: 'volcano-plot/volcano-plot.tpl.html',
-    controller: 'VolcanoPlotCtrl as vp'
-  };
+.component('volcanoPlot', {
```

Comment:
I'm really liking the simplification provided by `.component`. It's easy to see why this is "the way of the future" for Angular. 😎